Squashed commit of the following: (#97)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-tutorial.adoc

:page-aliases: start-project-task.adoc, apikit-create.adoc, apikit-run-test.adoc, apikit-add-logic.adoc, apikit-deploy.adoc, apikit-test-policy.adoc
:keywords: apikit, raml, tutorial, api, rest

In this workflow, you build an APIkit project that imports a RAML. You run the API in *Anypoint Studio* and use *API Console* to simulate calls to the API.

Studio 6.0 or later is recommended for this procedure. The API you create covers two main tasks:

* Updating vending machine data to update the central database on each sale
* Tracking inventory to find out if a machine is low on stock

== Create a New APIkit Project

This procedure launches the new project and generates a *Main Flow* with an *HTTP Listener*, an *APIkit Router*, and references to flows to handle errors. The project support REST resources and methods. These flows, called skeleton flows, are the bare minimum structure needed to support the RAML specification. The RAML file imported by this procedure is located at *src/main/api*.

. Download the example file to your local drive:
+
link:{attachmentsdir}/api.raml[api.raml]

> . Under the *File* menu, select *New* > *Mule Project*.

Reviewer: ```suggestion
. From the *File* menu, select *New* > *Mule Project*.
```

Reviewer: missed this one

Author: I really need to have accessibility words in my mind!!! Thanks!

. In the *New Mule Project* wizard, enter a *Project Name* (*myproject*, for example). *Mule Server 3.8.0* will be selected by default in the *Runtime* section.
+
. Check *Add APIkit Components*.
. In the *APIkit Settings* section, select the downloaded *api.raml* example file.
+
image::apikit-tutorial-3e0e4.png["Project Settings for a New Mule Project, including Project Name, Runtime, Maven Settings, Version Control System Support, APIkit Settings, and API Definition.",height=633,width=600]
+
. Click *Finish*.

== Run and Test the Project
. Right-click the project name in the *Project Explorer* and select *Run As* > *Mule Application*.
+

> The *APIkit Console* appears below the canvas, displaying the resource-action pairs exposed by the API.

Reviewer: ```suggestion
The *APIkit Console* appears and displays the resource-action pairs exposed by the API.
```

Reviewer: If possible, this removes non-accessible language

+

> . Click *Introduction* under the *Documentation* section to expand the API description.

Reviewer: ```suggestion
. Click *Introduction* to expand the API description.
```

> . Click *Introduction* under the *Documentation* section to expand the API description.

Reviewer: ```suggestion
. Click *Introduction* to expand the API description.
```

Reviewer: If possible, this removes non-accessible language

+
The main *Introduction* title and the sentence map to the contents of *Documentation* in the root section of the RAML file.
+

> . Click */machines* in the *Resource* section to expand *GET: /machines*.

Reviewer: ```suggestion
.  In the *Resource* section, click */machines* to expand *GET: /machines*.
```

. Click *Try It* label.
+

> The baseURI is optional in a RAML-backed API, but required by the API-Console *Try It* functionality; otherwise, *Try It* is disabled.

Reviewer: ```suggestion
The baseURI is optional in a RAML-backed API but is required by the the API-Console *Try It* functionality.  *Try It* is disabled if there is no baseURI.
```

Reviewer: Trying to get rid of the semicolon. check for accuracy ^

+

> image::fv-601.png["Example of the Try It feature being disabled because baseURI is not present."]

Reviewer: ```suggestion
image::fv-601.png["The Try It feature is disabled because baseURI is not present."]
```

+
In the RAML, you can use an arbitrary value, for example +http://www.example.com+ for the baseURI.
+

> . Click *GET* to send a test call to get a list of machines.

Reviewer: ```suggestion
.  To send a test call to get a list of machines, click *GET*.
```

== Add Logic to the Project

> For example purposes, add over-simplified logic to the project to use later in *API Manager*. Unlike the APIkit-generated flows, the flow you create during this procedure returns a result, a payload message, when you call the API from a browser.

Reviewer: ```suggestion
For example purposes, add over-simplified logic to the project to use later in *API Manager*. Unlike the APIkit-generated flows, the flow you create during this procedure returns  a payload message when you call the API from a browser.
```

. Drag an HTTP component to the canvas.
+
Studio creates a flow named *apiFlow* by default.
+
. In the *apiflow*, select the *HTTP Connector* to open its properties editor.

> . Click image:edit-16x16.gif[Edit button] to edit the *Connector Configuration* global configuration element.

Reviewer: ```suggestion
.  To edit the *Connector Configuration* global configuration element, click image:edit-16x16.gif[Edit].
```

Reviewer: does this really need to be an image?

Author: Right you are!
